### PR TITLE
E2E Appium - Initial setup, onboarding and import seed flow

### PR DESCRIPTION
**Pull request by Fatxx**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Implement new Appium, WebDriver.io and Cucumber e2e testing solution.
Adds initial test for "Create onboarding wizard flow"
Adds "Import wallet flow"

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: wdio/features/screen-objects/WalletSetupScreen.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { TERMS_AND_CONDITIONS_BUTTON_ID } from '../testIDs/Components/TermsAndConditions.testIds';
+import {
+  WALLET_SETUP_CREATE_NEW_WALLET_BUTTON_ID,
+  WALLET_SETUP_SCREEN_DESCRIPTION_ID,
+  WALLET_SETUP_SCREEN_IMPORT_FROM_SEED_BUTTON_ID,
+  WALLET_SETUP_SCREEN_TITLE_ID
+} from '../testIDs/Screens/WalletSetupScreen.testIds';
+
+class WalletSetupScreen {
```
Comment (cortisiko) [inline wdio/features/screen-objects/WalletSetupScreen.js:9]: > You should define your locator-id/test-id as a getter function then you perform the assertion on the element.

```
get walletScreenTitle() {
    return $(`~${WALLET_SETUP_SCREEN_TITLE_ID}`);
}

get walletScreenDescriptionText() {
    return $(`~${WALLET_SETUP_SCREEN_DESCRIPTION_ID}`);
}

  get importWalletButton() {
    return $(`~${WALLET_SETUP_SCREEN_IMPORT_FROM_SEED_BUTTON_ID}`);
  }

 get createWalletButton() {
    return $(`~${WALLET_SETUP_CREATE_NEW_WALLET_BUTTON_ID}`);
  }


async verifyScreenTitle(){
await expect(this.walletScreenTitle).toBeDisplayed();
}


async verifyImportWalletButton(){
await expect(this.importWalletButton).toBeDisplayed();

}

async verifyCreateNewWalletButton(){
await expect(this.createWalletButton).toBeDisplayed();

}


```

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  IMPORT_FROM_SEED_SCREEN_SUBMIT_BUTTON_ID,
+} from '../testIDs/Screens/ImportFromSeedScreen.testIds';
+import Selectors from '../helpers/Selectors';
+
+class ImportFromSeed {
+  async verifyScreenTitle() {
+    await expect(await $(`~${IMPORT_FROM_SEED_SCREEN_TITLE_ID}`)).toBeDisplayed();
+  }
+
+  async typeSecretRecoveryPhrase(phrase) {
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:15]: Same. re: utilizing gesture methods in the gestures.js file. it should be

```suggestion
 async typeSecretRecoveryPhrase(phrase) {
    await Gestures.typeText(IMPORT_FROM_SEED_SCREEN_SEED_PHRASE_INPUT_ID, phrase);
  }

```

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const elem = await $(`~${IMPORT_FROM_SEED_SCREEN_SEED_PHRASE_INPUT_ID}`)
+    await elem.setValue(phrase);
+  }
+
+  async typeNewPassword(newPassword) {
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:20]: Same. re: utilizing gesture methods in the gestures.js file. it should be

```suggestion
 async enterPassword(newPassword)  {
    await Gestures.typeText(IMPORT_FROM_SEED_SCREEN_NEW_PASSWORD_INPUT_ID, newPassword);
  }

```

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    const elem = await Selectors.getElementByPlatform(IMPORT_FROM_SEED_SCREEN_NEW_PASSWORD_INPUT_ID)
+    await elem.setValue(newPassword);
+  }
+
+  async typeConfirmPassword(confirmPassword) {
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:25]: Same. re: utilizing gesture methods in the gestures.js file. it should be

```suggestion
 async reEnterPassword(confirmPassword)  {
    await Gestures.typeText(IMPORT_FROM_SEED_SCREEN_CONFIRM_PASSWORD_INPUT_ID, confirmPassword);
  }

```

Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:25]: For some reason on android, the keyboard remains expanded so you would need to collapse the keyboard. Tapping on any element outside the input box should work.

**Review by cortisiko — CHANGES_REQUESTED** [review]
Hey @Fatxx I left some  feedback comments, lmk if you have any questions!

File: wdio/features/screen-objects/OptinMetricsScreen.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { OPTIN_METRICS_I_AGREE_BUTTON_ID, OPTIN_METRICS_TITLE_ID } from '../testIDs/Screens/OptinMetricsScreen.testIds';
+
+class OptinMetricsScreen {
+  async verifyScreenTitle() {
+    await expect(await $(`~${OPTIN_METRICS_TITLE_ID}`)).toBeDisplayed();
+  }
+
+  async clickIAgreeButton() {
+    const elem = await $(`~${OPTIN_METRICS_I_AGREE_BUTTON_ID}`)
```
[the remark this answers is not in the document]
Comment (Fatxx) [reply wdio/features/screen-objects/OptinMetricsScreen.js:9]: Here I'm waiting for the button to disappear, not to appear before click

File: wdio/features/screen-objects/WalletMainScreen.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { ONBOARDING_WIZARD_STEP_1_CONTAINER_ID } from '../testIDs/Components/OnboardingWizard.testIds';
+
+class WalletMainScreen {
```
[the remark this answers is not in the document]
Comment (Fatxx) [reply wdio/features/screen-objects/WalletMainScreen.js:3]: This is the actual Wallet Main Screen after you complete the onboarding

**Pull request closed without merging**